integration_tests/src/main/python/string_test.py

```diff
@@ -320,15 +320,15 @@ def test_initcap():
     # the charicter set to something more reasonable
     # upper and lower should cover the corner cases, this is mostly to
     # see if there are issues with spaces
-    gen = mk_str_gen('([aAbB1357_@%-]{0,12}[ \r\n\t]{1,2}){1,5}')
+    gen = mk_str_gen('([aAbB1357ȺéŸ_@%-]{0,15}[ \r\n\t]{1,2}){1,5}')
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark: unary_op_df(spark, gen).select(
                 f.initcap(f.col('a'))))
 
 @incompat
-@pytest.mark.xfail(reason='https://github.com/rapidsai/cudf/issues/8644')
-def test_initcap_width_change():
-    gen = mk_str_gen('ŉ([aAbB13ŉȺéŉŸ]{0,5}){1,5}')
+@pytest.mark.xfail(reason='Spark initcap will not convert ŉ to ʼN')
```

Comment:
This is fixed by https://github.com/rapidsai/cudf/pull/8647 and should no longer be xfailed.  It's also weird to mark a test as xfailed but then not test the condition that fails?

Reply:
This test is still failing. 
Because of the fix (rapidsai/cudf#8647), GPU initcap will convert the first  `ŉ` to `ʼN`, but CPU initcap will not, as the reason string said.

Log snip from the pre-merge build.
```Bash
[2021-07-05T08:22:24.286Z]   opposite null order not supported
[2021-07-05T08:22:24.286Z] XFAIL ../../src/main/python/string_test.py::test_initcap_special_chars[INCOMPAT]
[2021-07-05T08:22:24.286Z]   Spark initcap will not convert ŉ to ʼN
```